### PR TITLE
Avoid port conflicts in the networking tests.

### PR DESCRIPTION
**Pull request by urbanserj**
## High-level description

Marathon randomly assigns host ports. In rare cases it causes port conflicts in the networking integration tests. We shouldn't rely on this feature and have to assign unique port numbers manually.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS-46220](https://jira.mesosphere.com/browse/DCOS-46220) test_networking.test_vip can fail because Marathon says Constraints for run spec [xxx] not satisfied.

## Related tickets (optional)

Other tickets related to this change:

  - [DCOS_OSS-3736](https://jira.mesosphere.com/browse/DCOS_OSS-3736) Update marathon app configuration to match Marathon 1.5 syntax

## Checklist for all PRs

  - [x] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change: integration tests
  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: none
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by urbanserj** [comment]
@mesosphere-mergebot bump-ee

**Comment by urbanserj** [comment]
@mesosphere-mergebot label [Ready For Review]

File: packages/dcos-integration-test/extra/test_helpers.py [hunk 1]
```diff
@@ -85,9 +78,9 @@ def marathon_test_app_linux(
         'instances': 1,
         'cmd': '/opt/mesosphere/bin/dcos-shell python '
                '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py {}'.format(
-                   # If network is host and host port is zero, then the port is auto-assigned
-                   # and the commandline should reference the port with the marathon built-in
-                   '$PORT0' if host_port == 0 and network == marathon.Network.HOST else container_port),
+                   # If container port is not defined, then the port is auto-assigned and
+                   # the commandline should reference the port with the marathon built-in
+                   '$PORT0' if container_port is None else container_port),
```
[the remark this answers is not in the document]
Comment (urbanserj) [reply packages/dcos-integration-test/extra/test_helpers.py]: That's correct, in that case marathon will set `$PORT0` to the value of `hostPort`.

**Comment by urbanserj** [comment]
@mesosphere-mergebot bump-ee

File: packages/dcos-integration-test/extra/test_networking.py [hunk 2]
```diff
@@ -85,13 +78,11 @@ def info(self, dcos_api_session):
...
         info = self.info(dcos_api_session)
         task = info['app']['tasks'][0]
-        if self._network == marathon.Network.USER:
-            if self._container == marathon.Container.DOCKER:
-                host = task['host']
-                port = task['ports'][0]
-            else:
-                host = task['ipAddresses'][0]['ipAddress']
-                port = self.app['ipAddress']['discovery']['ports'][0]['number']
+        if 'networks' in self.app and \
```
[the remark this answers is not in the document]
Comment (urbanserj) [reply packages/dcos-integration-test/extra/test_networking.py]: I'm sorry I didn't quite understand your question, but let me explain the logic of this code.

In test_vip, we start `proxy` and `origin` applications on host, bridge and overlay networks and check connectivity between them. For that, we send a request to `proxy` app that sends another request to `origin` app using its vip.

For the first http request, we should find `proxy` host and ip somehow. For containers on the `host`, `bridge` and `dcos6` overlay networks, we use a pair of agent ip and host or port-mapped port. For containers on `dcos` overlay network, we don't have a port mapping, would like not to have a port mapping, and connect to them directly using overlay ip and container port. We don't care about the runtime here, that's why that code was changed.

I didn't find any evidence about the deprecation of `ipAddress` in marathon response, as to [1][2] it was only deprecated in the app definition. Also there is no other way to get container ip using marathon api.

[1] http://mesosphere.github.io/marathon/docs/upgrade/network-api-migration.html
[2] http://mesosphere.github.io/marathon/docs/upgrade/network-api-for-apps.html

Please let me know if you have any more questions.

File: packages/dcos-integration-test/extra/test_networking.py [hunk 3]
```diff
@@ -439,7 +420,7 @@ def test_ip_per_container(dcos_api_session):
 
     with dcos_api_session.marathon.deploy_and_cleanup(app_definition, check_health=True):
         service_points = dcos_api_session.marathon.get_app_service_endpoints(app_definition['id'])
-        app_port = app_definition['container']['docker']['portMappings'][0]['containerPort']
+        app_port = app_definition['container']['portMappings'][0]['containerPort']
```
[the remark this answers is not in the document]
Comment (mainred) [reply packages/dcos-integration-test/extra/test_networking.py]: But the line I pointed is host_port
`host_port=9080`

**Pull request merged**